### DISCUSSION TITLE
Post action using REST to Hasura database

### DISCUSSION THREAD

**Discussion by PhilipOyelegbin** [hasura/graphql-engine#8823] [Question] [answered]
I'm currently having issues posting  my form data to my database from the frontend, please help
![code](https://user-images.githubusercontent.com/72349309/185644155-b7e44dcb-a3e8-4144-b82f-6f8647cd484e.png)
![error](https://user-images.githubusercontent.com/72349309/185644249-7bc00f61-2b72-4346-9a2f-0a79b6d1ea4f.png)

**Comment by meetzaveri** [comment, marked as answer]
Hello Philip,

Can you send us the message that error shows? Generally, `400` error code is thrown when there is invalid syntax(JSON payload) or malformed request body. 

>In the case of a REST API with a JSON payload, 400's are typically, and correctly I would say, used to indicate that the JSON is invalid in some way according to the API specification for the service.

## Related Solution

### You need to use `mode:"cors"` when you are passing `Content-type:"application/json"`
The `no-cors` mode only works with some constraint. Covering from [mdn docs regarding fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options)
>Note that mode: "no-cors" only allows a limited set of headers in the request:

I check that you've set request mode to `no-cors` in fetch options. You can only use `mode:no-cors` when you pass specific limited set of headers in request as mentioned in that mdn docs.

here's the common `options` object for fetch API
```js
 method: "post", // *GET, POST, PUT, DELETE, etc.
 mode: "cors", // no-cors, *cors, same-origin
 headers: {
    "Content-Type": "application/json"
 },
 body: JSON.stringify(data) // body data type must match "Content-Type" header
```

### Example 

I created a simple app via codesandbox with react client. Created a REST endpoint from my hasura API explorer and used it to insert a sample data. That REST endpoint contains mutation to insert data into db.

Here's the codesandbox link you can test - https://codesandbox.io/s/mystifying-platform-97wdks?file=/src/App.js . You can also access the preview app for that codesandbox here - https://97wdks.csb.app/


Do let us know further if you are experiencing any problems, thanks!

**Reply by PhilipOyelegbin** [reply]
Below is the error code I'm getting, I think it might be from how I set up the database on the Hasura cloud. I'm currently searching for a video to verify my configuration for the post request, if you know of any video that can help please kindly recommend it.

![error](https://user-images.githubusercontent.com/72349309/186176448-2d3cc1f5-c309-480b-84db-3951ed832978.png)

**Reply by meetzaveri** [reply]
Got it. Did you try changing your API options as suggested in the answer above ?

**Reply by PhilipOyelegbin** [reply]
Yes I did, I even tested my code with a demo API "webhook" and it worked but the Hasura API did not work. I noticed the error message is referring me back to the url in my fetch which is why I need a video that explains the concept of setting up a REST url for post action on Hasura. I might have set it up wrongly.
